internal/engine/watchmanager.go

```diff
@@ -86,9 +86,26 @@ func (w *WatchManager) diff(ctx context.Context, st store.RStore) (setup []Watch
 		manifestsToProcess[ConfigsManifestName] = &configsManifest{dependencies: append([]string(nil), state.ConfigFiles...)}
 	}
 
-	for name := range w.manifestWatches {
-		if _, ok := manifestsToProcess[name]; !ok {
+	for name, mnc := range w.manifestWatches {
+		m, ok := manifestsToProcess[name]
+		if !ok {
+			teardown = append(teardown, name)
+			continue
+		}
+
+		if len(m.Dependencies()) != len(mnc.manifest.Dependencies()) {
```

Comment:
can you break this out into a helper function that checks if the dependencies match?